### DISCUSSION TITLE
Deprecation package updates

### DISCUSSION THREAD

**Discussion by florentdemelezi** [aquacropos/aquacrop#134] [Q&A]
Hi all,

I am trying to use aquacrop in vscode although using python 3.8.10 and 3.10 I seem to get the same error message for 
from numba.pycc import CC
which prohibits running the modules and package itself in visual-studio-code.
Is it planned to update the aquacrop module or if someone solved it they could guide me through it?
I noticed that the **from numba.pycc import CC** is deprecated and need to be modified?
# temporary name for compiled module
cc = CC("solution_cc_development")

@cc.export("cc_development", "f8(f8,f8,f8,f8,f8,unicode_type,f8)")
def cc_development(
    CCo: float,

**Comment by SongshGeo** [comment]
I'm facing the same problem without any available idea.

**Comment by NickAlexander18** [comment]
I guess the version of numba play an important role. The numba>0.57.0 will deprecate the pycc module, you can use older version such as 0.56.0. However, I think the actual problem may not be the numba. 

![image](https://github.com/aquacropos/aquacrop/assets/52007523/18cf0e1d-f6a3-4446-997d-04bbfa9eaeff)

**Comment by tfoster88** [comment]
It seems to be a combination of Python version, numba version, and other dependencies that cause this issue. We are going to look into implementing an alternative to numba AOT over the coming months, but in the meantime the temporary fix is to run in development environment (assuming you have already ensured you have a suitable compiler installed on your machine). This will increase run time, but the numba AOT that we originally implemented to speed things up has become increasingly difficult to get to work stably with AquaCrop-OSPy and I'm going to need a decent chunk of time to dig into why (and alternatives)
